cmd/syft/cli/attest/attest.go

```diff
@@ -183,23 +209,72 @@ func generateAttestation(predicate []byte, src *source.Source, sv *sign.SignerVe
 		return errors.Wrap(err, "unable to sign SBOM")
 	}
 
+	if app.Attest.NoUpload {
+		bus.Publish(partybus.Event{
+			Type: event.Exit,
+			Value: func() error {
+				_, err := os.Stdout.Write(signedPayload)
+				return err
+			},
+		})
+	}
+
+	return uploadAttestation(app, signedPayload, digest, sv)
+}
+
+func uploadAttestation(app *config.Application, signedPayload []byte, digest name.Digest, sv *sign.SignerVerifier) error {
+	opts := []static.Option{static.WithLayerMediaType(types.DssePayloadType)}
+	if sv.Cert != nil {
+		opts = append(opts, static.WithCertChain(sv.Cert, sv.Chain))
+	}
+
+	bundle, err := uploadToTlog(context.TODO(), sv, app.Attest.RekorURL, func(r *client.Rekor, b []byte) (*models.LogEntryAnon, error) {
+		return cosign.TLogUploadInTotoAttestation(context.TODO(), r, signedPayload, b)
+	})
+	if err != nil {
+		return err
+	}
+	opts = append(opts, static.WithBundle(bundle))
+	sig, err := static.NewAttestation(signedPayload, opts...)
+	if err != nil {
+		return err
+	}
+
+	se, err := ociremote.SignedEntity(digest)
+	if err != nil {
+		return err
+	}
+
+	// Attach the attestation to the entity.
+	newSE, err := mutate.AttachAttestationToEntity(se, sig)
+	if err != nil {
+		return err
+	}
+
+	// Publish the attestations associated with this entity
+	err = ociremote.WriteAttestations(digest.Repository, newSE)
+	if err != nil {
+		return err
+	}
+
 	bus.Publish(partybus.Event{
 		Type: event.Exit,
 		Value: func() error {
 			_, err := os.Stdout.Write(signedPayload)
```

Comment:
if we're uploading should we be outputting the signed payload to the screen? or suppress this?

Reply:
I think we should still be publishing the payload to the screen on upload so the user is able to inspect what was uploaded, but I can also see how suppressing is gives a cleaner execution and we don't pollute the screen since payloads are large and noticeably indecipherable on first glance

Reply:
cc @luhring for input

Reply:
Yeah no strong opinion, but I think I'd lean toward not outputting it for now for the "clean feel" reason

Reply:
removed output on upload